packages/connect-core/src/entities/App.ts

```diff
@@ -73,11 +78,18 @@ export default class App extends CoreEntity {
     data: AppData,
     connector: ConnectorInterface
   ): Promise<App> {
-    const artifact: AragonArtifact = await resolveMetadata(
-      'artifact.json',
-      data.contentUri || undefined,
-      data.artifact
-    )
+    let artifact: AragonArtifact
+    if (hasAppInfo(data.appId, 'apm')) {
+      artifact = getApmInternalAppInfo(data.appId)
+    } else if (hasAppInfo(data.appId, 'aragon')) {
+      artifact = getAragonOsInternalAppInfo(data.appId)
+    } else {
+      artifact = await resolveMetadata(
+        'artifact.json',
+        data.contentUri || undefined,
+        data.artifact
+      )
+    }
```

Comment:
I was wondering if we should move this in a utility function? (Probably something we can do later)

Reply:
Yes, good idea!